### PR TITLE
Exp - 2 code review

### PR DESCRIPTION
**Pull request by srisatyavinay**

### CODE REVIEW

File: experiment/simulation/js/circuit.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+                        if (listPmos[j].voltage === 0) {
+                            listPmos[j].outvoltage = 9;
+                        }
+                        else {
+                            listPmos[j].outvoltage = -5;
+                        }
+
+                        //listPmos[j].outvoltage = 5;
+                    }
+                    console.log("Voltage stored in pmos");
```
Comment (vjspranav) [inline experiment/simulation/js/circuit.js:70]: Remove the console log statements wherever present.

File: experiment/simulation/js/circuit.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+            const r = listNmos[i].id.concat("$", listOutput[j].id);
+            if (connectionMap.has(r)) {
+                // if nmos 
+                if (listNmos[i].outterminal === 1) {
+                    listOutput[j].voltage = listNmos[i].outvoltage;
+                    nmosNand++;
+                    console.log("Nmos succesfully conducted");
+                }
+
+
```
Comment (vjspranav) [inline experiment/simulation/js/circuit.js:160]: Please use prettier code formatter, and format the javascript (for all experiments pushed)

File: experiment/simulation/js/integrate.js [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+    d1.outputsign = 1;
+    const container = document.getElementById("diagram");
+
+
+    container.insertAdjacentElement("afterbegin", d);
+
+    jsplumbInstance.draggable(id, { "containment": true });
+    listOutput.push(d1);
+
+    addInstanceFinalOutput(id);
```
Comment (vjspranav) [inline experiment/simulation/js/integrate.js:328]: This file, has almost no comments and the variable names in a few places are almost making it impossible to understand what the code is doing. 
Please add some comments, and use better naming

File: experiment/simulation/js/nand.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+function checkNor() {
+    const x = permutator([0, 1])
+    let norCircuitValid = 0
+    for (let i = 0; i < x.length; i++) {
+        for (let j = 0; j < x.length; j++) {
+            for (let k = 0; k < x.length; k++) {
+                if (connectionMap.has("input" + x[i][0] + "$pmos" + x[j][0]) && connectionMap.has("input" + x[i][1] + "$pmos" + x[j][1]) && connectionMap.has("input" + x[i][0] + "$nmos" + x[k][0]) && connectionMap.has("input" + x[i][1] + "$nmos" + x[k][1]) && connectionMap.has("vdd0$pmos" + x[j][0]) && connectionMap.has("ground0$nmos" + x[k][0]) && connectionMap.has("ground0$nmos" + x[k][1]) && connectionMap.has("pmos" + x[j][1] + "$output0") && connectionMap.has("nmos" + x[k][1] + "$output0") && connectionMap.has("nmos" + x[k][0] + "$output0") && connectionMap.has("pmos" + x[j][0] + "$pmos" + x[j][1]) && (connectionMap.size === 11)) {
+                    norCircuitValid = 1
+                }
```
Comment (vjspranav) [inline experiment/simulation/js/nand.js:41]: Add a break here too, why let the loop run extra?

File: experiment/simulation/js/nand.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+function changeObservation(htmlText, removedClass, addedClass) {
+    document.getElementById("output-box").innerHTML = htmlText
+    document.getElementById("output-box").classList.remove(removedClass)
+    document.getElementById("output-box").classList.add(addedClass)
+}
+
+function showTruthTable() {
+    const output = [0, 0, 0, 0];
+    a = document.getElementById("table-body")
```
Comment (vjspranav) [inline experiment/simulation/js/nand.js:118]: Don't initialise without let or const

**Review by vjspranav — CHANGES_REQUESTED** [review]

File: experiment/simulation/js/circuit.js (hunk 1 above)
Comment (srisatyavinay) [reply experiment/simulation/js/circuit.js:70]: Removed them

File: experiment/simulation/js/circuit.js (hunk 2 above)
Comment (srisatyavinay) [reply experiment/simulation/js/circuit.js:160]: Used it and formatted evrything

File: experiment/simulation/js/integrate.js (hunk 3 above)
Comment (srisatyavinay) [reply experiment/simulation/js/integrate.js:328]: Wrote some comments already in exp - 1. It's the same here.

File: experiment/simulation/js/nand.js (hunk 4 above)
Comment (srisatyavinay) [reply experiment/simulation/js/nand.js:41]: Added break now

File: experiment/simulation/js/nand.js (hunk 5 above)
Comment (srisatyavinay) [reply experiment/simulation/js/nand.js:118]: Initialised every variable using let or const

**Pull request closed without merging**